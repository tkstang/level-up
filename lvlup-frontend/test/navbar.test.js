import NavBar from '../src/components/Navbar/navbar';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import lvlupApp from '../src/reducers/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('NavBar should render', () => {
  const store = mockStore({ lvlupApp });
  const nav = shallow(
    <NavBar studentLoginInfo={{ status: true }} adminLoginInfo={{ status: true }} store={store} />,
  );
  expect(shallowToJson(nav)).toMatchSnapshot();
});
