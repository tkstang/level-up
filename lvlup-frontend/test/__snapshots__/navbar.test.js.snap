exports[`test NavBar should render 1`] = `
<Menu
  className="nav"
  secondary={true}
  size="small">
  <MenuItem
    className="center"
    header={true}>
    <Link
      replace={false}
      to="/student/dashboard">
      <div
        className="nav-logo">
        lvl^
      </div>
    </Link>
    <Link
      replace={false}
      to="/admin/dashboard">
      <div
        className="nav-logo">
        lvl^
      </div>
    </Link>
  </MenuItem>
  <MenuItem
    className="right">
    <Connect(LogOutGithub) />
    <Connect(LogOutAdmin) />
  </MenuItem>
</Menu>
`;
